pkg/labels/array.go

```diff
@@ -113,24 +113,17 @@ nextLabel:
 	return missing
 }
 
-// Has returns whether the provided key exists.
+// Has returns whether the provided key exists in the label array.
 // Implementation of the
 // github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/labels.Labels interface.
 func (ls LabelArray) Has(key string) bool {
 	// The key is submitted in the form of `source.key=value`
 	keyLabel := parseSelectLabel(key, '.')
-	if keyLabel.IsAnySource() {
```

Comment:
Do we have good unit tests for the "any source" handling of this? Combinations of selectors selecting "any" source and matching keys, or selectors with specific sources and matching/nonmatching sources for the key?

Seeing the code for source handling getting dropped in a commit that is specifically about one special source and its own keys makes me a little bit nervous.

Reply:
We do now; I added some tests *first*, then refactored. So we should be good here.